cookies/resources/testharness-helpers.js

```diff
@@ -0,0 +1,42 @@
+// Given an array of potentially asynchronous tests, this function will execute
+// each in serial, ensuring that one and only one test is executing at a time.
+//
+// The test array should look like this:
+//
+//
+//     var tests = [
+//       [
+//         "Test description goes here.",
+//         function () {
+//           // Test code goes here. `this` is bound to the test object.
+//         }
+//       ],
+//       ...
+//     ];
+//
+// The |setup| and |teardown| arguments are functions which are executed before
+// and after each test, respectively.
+function executeTestsSerially(testList, setup, teardown) {
```

Comment:
My recommended way of scheduling tests to run in serial is to use `add_result_callback` to deal with scheduling the next test, so that `test.add_cleanup` is only used for post-test cleanup, as intended.

There isn't a general high-level construct for this in testharness.js and it isn't entirely clear to me if all the use cases are similar enough to warrant one (e.g. in this case it seems you rely on the fact that the test will only have a single step).